browser/js/app.js

```diff
@@ -52,4 +52,12 @@ app.run(function ($rootScope, AuthService, $state) {
 
     });
 
+    // The code below will give descriptive errors if there are issues with resolve in the states 
+
+    $rootScope.$on('$stateChangeError', function(event, toState, toParams, fromState, fromParams, error){ 
+        // this is required if you want to prevent the $UrlRouter reverting the URL to the previous valid location
+        event.preventDefault();
+        $log.error(error);
+    });
+
```

Comment:
This allows angular to log errors even when they happen during the resolve of states. Might be helpful.

reference, last answer: http://stackoverflow.com/questions/25962417/how-to-handle-error-in-angular-ui-routers-resolve